source/libs/api.ts

```diff
@@ -67,15 +74,16 @@ function fetch4(query, personalToken) {
 }
 
 // Main function: handles cache, options, errors
-async function call(fetch, query, options = {accept404: false}) {
+async function call<T extends SoManyFetch>(fetch: T, query: string, options = {accept404: false}) {
 	if (cache.has(query)) {
 		return cache.get(query);
 	}
 
-	const {personalToken} = await new OptionsSync().getAll();
-	const response = await fetch(query, personalToken);
+	const {personalToken} = await new OptionsSync().getAll<{personalToken: string}>();
+	const response = await fetch4(query, personalToken);
 	const content = await response.text();
-	const result = content.length > 0 ? JSON.parse(content) : {status: response.status};
+
+	const result: { data?: unknown, errors?: Error[], message?: string} = content.length > 0 ? JSON.parse(content): {};
```

Comment:
I put all the properties optional, because originally, if the `content.length > 0`, you were returning `{ status: response.status }` which is never used, so I simply return an empty object now.